### PR TITLE
[PostgreSQL] Built-in functions, merge statement and Non Ansi Join

### PR DESCRIPTION
**Pull request by ealight**
- Added all the mathematics built-in functions. ([Source](https://www.postgresql.org/docs/15/functions-math.html))
- Added all the string built-in functions. ([Source](https://www.postgresql.org/docs/15/functions-string.html))
- Added all the date built-in functions. ([Source](https://www.postgresql.org/docs/15/functions-datetime.html))
- Added all the date formatting built-in functions. ([Source](https://www.postgresql.org/docs/15/functions-formatting.html))
- Added MERGE statement handler([Source](https://www.postgresql.org/docs/current/sql-merge.html))
- Added old fashion Non Ansi Join handler ([Source](https://sqlenlight.com/support/help/sa0006/))

### CODE REVIEW

**Comment by ealight** [comment]
Please, squash the commits when merging.

File: sql/postgresql/PostgreSQLParser.g4 [hunk 1]
```diff
@@ -1926,8 +1928,8 @@ func_return
 
 func_type
    : typename
-   | type_function_name attrs PERCENT TYPE_P
-   | SETOF type_function_name attrs PERCENT TYPE_P
+   | (builtin_function_name | type_function_name) attrs PERCENT TYPE_P
+   | SETOF (builtin_function_name | type_function_name) attrs PERCENT TYPE_P
```
Comment (KvanTTT) [inline sql/postgresql/PostgreSQLParser.g4:1932]: Please simplify:

```suggestion
   | SETOF? (builtin_function_name | type_function_name) attrs PERCENT TYPE_P
```

Comment (ealight) [reply sql/postgresql/PostgreSQLParser.g4]: Done

**Comment by ealight** [comment]
@KvanTTT can you please merge it? Thank you.

**Pull request merged**